R/readHouseholdSurveys.R

```diff
@@ -12,24 +12,48 @@
 #' @importFrom quitte as.quitte
 #' @importFrom magclass as.magpie
 
-readHouseholdSurveys <- function(subtype = c("cooking", "lighting")) {
+readHouseholdSurveys <- function(subtype = c("cooking", "lighting", "appliances")) {
 
   # read data sheet
   data <- switch(subtype,
-                 cooking  = read.xlsx("Cooking_fuels.xlsx"),
-                 lighting = read.xlsx("Lighting_fuels.xlsx"))
+                 cooking    = read.xlsx("Cooking_fuels.xlsx"),
+                 lighting   = read.xlsx("Lighting_fuels.xlsx"),
+                 appliances = read.xlsx("Appliances.xlsx"))
+
+  # region mapping
+  regionmap <- toolGetMapping("regionmappingHouseholdSurveys.csv",
+                              type = "regional",
+                              where = "mredgebuildings")
 
   # standardize column names
   colnames(data) <- tolower(colnames(data))
 
   # change country codes
   data <- data %>%
-    rename("period" = "year",
-           "region" = "country_long") %>%
-    pivot_longer(cols = -all_of(c("region", "period")), names_to = "carrier", values_to = "value") %>%
-    mutate(carrier = sub("\\.", "_", .data$carrier),
-           enduse = subtype,
-           period = ifelse(is.na(.data$period), median(.data$period, na.rm = TRUE), .data$period)) %>%
+    mutate(region = toolCountry2isocode(country = .data$country_long,
+                                        mapping = setNames(as.list(regionmap$regionTarget),
+                                                           regionmap$region))) %>%
```

Comment:
Stuff like this should usually happen in the convert function. But you can keep it here this time if it requires too much effort to change it.